test/e2e/network/conntrack.go

```diff
@@ -268,13 +269,136 @@ var _ = SIGDescribe("Conntrack", func() {
+			"app": "boom-server",
+		}
+
+		serverPod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   "boom-server",
+				Labels: serverLabel,
+			},
+			Spec: v1.PodSpec{
+				NodeName: serverNodeInfo.name,
+				Containers: []v1.Container{
+					{
+						Name:  "boom-server",
+						Image: "aojea/conntrack:invalid",
+						Ports: []v1.ContainerPort{
+							{
+								ContainerPort: 9000, // Default port exposed by boom-server
+							},
+						},
+						Env: []v1.EnvVar{
+							{
+								Name: "POD_IP",
+								ValueFrom: &v1.EnvVarSource{
+									FieldRef: &v1.ObjectFieldSelector{
+										APIVersion: "v1",
+										FieldPath:  "status.podIP",
+									},
+								},
+							},
+							{
+								Name: "POD_IPS",
+								ValueFrom: &v1.EnvVarSource{
+									FieldRef: &v1.ObjectFieldSelector{
+										APIVersion: "v1",
+										FieldPath:  "status.podIPs",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		_, err := fr.ClientSet.CoreV1().Pods(fr.Namespace.Name).Create(context.TODO(), serverPod, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		err = e2epod.WaitForPodsRunningReady(fr.ClientSet, fr.Namespace.Name, 1, 0, framework.PodReadyBeforeTimeout, map[string]string{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Server pod created on node " + serverNodeInfo.name)
+
+		svc := &v1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "boom-server",
+			},
+			Spec: v1.ServiceSpec{
+				Selector: serverLabel,
+				Ports: []v1.ServicePort{
+					{
+						Protocol: v1.ProtocolTCP,
+						Port:     9000,
+					},
+				},
+			},
+		}
+		_, err = fr.ClientSet.CoreV1().Services(fr.Namespace.Name).Create(context.TODO(), svc, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Server service created")
+
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "startup-script",
+			},
+			Spec: v1.PodSpec{
+				NodeName: clientNodeInfo.name,
+				Containers: []v1.Container{
+					{
+						Name:  "startup-script",
+						Image: imageutils.GetE2EImage(imageutils.BusyBox),
+						Command: []string{
+							"sh", "-c", "while true; do sleep 2; nc boom-server 9000& done",
+						},
+					},
+				},
+				RestartPolicy: v1.RestartPolicyNever,
+			},
+		}
+		_, err = fr.ClientSet.CoreV1().Pods(fr.Namespace.Name).Create(context.TODO(), pod, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Client pod created")
+
+		// The client will open connections against the server
+		// The server will inject invalid packets
+		// if conntrack does not drop the invalid packets it will go through without NAT
+		// so the client will receive an unexpected TCP connection and RST the connection
+		// the server will log ERROR if that happens
+		ginkgo.By("checking client pod does not RST the TCP connection because it receives and INVALID packet")
+		if err := wait.PollImmediate(5*time.Second, time.Minute, logContainsFn("ERROR", "boom-server")); err == nil {
```

Comment:
the test fails if ERROR string is in the logs, that means that we've received a RST

Reply:
why checking log, the pod doesn't terminate anymore?

Reply:
the pod can terminate by different reasons, is not usual, but it happens in the CI. 
This verifies exactly the behaviour that we want to test, that no RST is received

Reply:
+1